docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,474 @@
+The process of the contract giving up residence on the origin DAML ledger and taking up residence on the target DAML ledger is called a **contract transfer**.
+The **Enter** and **Leave** events on the transaction stream originate from such contract transfers, as will be explained below.
+Moreover, contract transfers are synchronization points between the origin and target DAML ledgers and therefore affect the ordering guarantees.
+We therefore generalize causality graphs for interoperability.
+
+Definition »Transfer action«
+  A **transfer action** on a contract `c` is written **Transfer** `c`.
+  The **informees** of the transfer actions are the stakeholders of `c`.
+
+In the following, the term *action* refers to transaction actions (**Create**, **Exercise**, **Fetch**, and **NoSuchKey**) as well as transfer actions.
+In particular, a transfer action on a contract `c` is an action on `c`.
+Transfer actions do not appear in transactions though.
+So a transaction action cannot have a transfer action as a consequence and transfer actions do not have consequences at all.
+
+Definition »Interoperable causality graph«
+  An **interoperable causality graph** `G` for a set `Y` of DAML ledgers is a finite, transitively closed, directed acyclic graph.
+  The vertices are either transactions or transfer actions.
+  Every action is possibly annotated with an **incoming ledger** and an **outgoing ledger** from `Y` according to the following table:
+
+  +---------------+-----------------+-----------------+
+  | Action        | incoming ledger | outgoing ledger |
+  +===============+=================+=================+
+  | **Create**    | no              | yes             |
+  +---------------+-----------------+-----------------+
+  | consuming     |                 |                 |
+  | **Exercise**  | yes             | no              |
+  +---------------+-----------------+-----------------+
+  | non-consuming |                 |                 |
+  | **Exercise**  | yes             | yes             |
+  +---------------+-----------------+-----------------+
+  | **Fetch**     | yes             | yes             |
+  +---------------+-----------------+-----------------+
+  | **NoSuchKey** | no              | no              |
+  +---------------+-----------------+-----------------+
+  | **Transfer**  | maybe           | maybe           |
+  +---------------+-----------------+-----------------+
+
+  For non-consuming **Exercise** and **Fetch** actions, the incoming ledger must be the same as the outgoing ledger.
+  **Transfer** actions must have at least one of them.
+  A **transfer** action with both set represents a complete transfer. If only the incoming ledger is set, that represents the partial information of an **Enter** event, if only outgoing is set, that's the partial information of a **Leave** event.
```

Comment:
In Canton, we don't have atomic transfers in reality. What's the benefit of having them in the model? I know that you want to be able to "cancel out" successive leaves and enters locally at participants, but I don't think this really allows you to do that, as they can't be atomic on the "global ledger".

Reply:
As discussed, we'll leave it this way as it's anyway marked as WIP. We can revisit in a few months when the situation is clearer.